tests/Core.nut

```diff
@@ -0,0 +1,422 @@
+                                    }
+                                } else {
+                                    fail("Response verification failed by params.callbackVerify function");
+                                }
+                            } catch (ex) {
+                                fail("Unexpected error while send request (sendasync): " + ex);
+                            }
+                        }.bindenv(this));
+                    }
+                } catch (ex) {
+                    fail("Unexpected error while send request: " + ex);
+                }
+            }.bindenv(this));
+        } catch (ex) {
+            fail("Unexpected error while send request: " + ex);
+            return;
+        }
+    }.bindenv(this));
+}
+
+// createTestAll
+// Create Promise for series of tests testing Rocky
+// 
+// @param {array} tests - Array of 'params' for createTest(params)
+// @param {string} type - The condition for the success of all tests (only_successes|only_fails)
+// @return {Promise}
+function createTestAll(tests = [], type = "only_successes") {
+    return Promise(function(resolve, reject) {
+        try {
+            if (typeof tests != "array") {
+                throw "Invalid type of createTestAll 'tests': " + typeof tests;
+            }
+            local length = tests.len();
+            local index = 0;
+            local successes = 0;
+            local fails = 0;
+            local last_reason = null;
+            local execute;
+            execute = function() {
+                try {
+                    switch (type) {
+                        case "only_fails": {
+                            if (successes > 0) {
+                                reject("createTestAll resolved one of the tests, but 'type' was 'only_fails'");
+                                return;
+                            }
+                            if (fails >= length) {
+                                resolve();
+                                return;
+                            }
+                            break;
+                        }
+                        case "only_successes": default: {
+                            if (fails > 0) {
+                                reject(last_reason);
+                                return;
+                            }
+                            if (successes >= length) {
+                                resolve();
+                                return;
+                            }
+                            break;
+                        }
+                    }
+                    local test = tests[index++];
+                    if (typeof test != "table") {
+                        throw "Invalid type of createTest 'params': " + typeof test;
+                    }
+                    local expect = "success";
+                    if ("_createTestExpect" in test) {

```

Comment:
The createTestAll function takes an array of tables for sequential execution by the createTest function. So, to take into account the second parameter (expect) of the createTest function, each table in the array may contain the parameter '_createTestExpect', which allows you to specify the 'expect' parameter of the createTest function.